ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/authorization/AuthorizationInterceptor.scala

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.server.api.authorization
+
+import com.daml.ledger.participant.state.v1.{AuthService, Claims}
+import io.grpc.{
+  Context,
+  Contexts,
+  Metadata,
+  ServerCall,
+  ServerCallHandler,
+  ServerInterceptor,
+  Status
+}
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.compat.java8.FutureConverters
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+/**
+  * This interceptor uses the given [[AuthService]] to get [[Claims]] for the current request,
+  * and then stores them in the current [[Context]].
+  *
+  * Use [[ApiServiceAuthorization]] to read the claims from the context.
+  * */
+class AuthorizationInterceptor(protected val authService: AuthService, ec: ExecutionContext)
```

Comment:
I'd prefer to keep class hierarchies closed, unless we have a valid use case for inheritance.
```suggestion
final class AuthorizationInterceptor(authService: AuthService, ec: ExecutionContext)
```